middleware/admin.js

```diff
@@ -35,50 +38,63 @@ function adminLogout (request, response, keycloak) {
   });
 
   request.on('end', function () {
-    let payload;
-    let parts = data.split('.');
+    let token = new Token(data);
+    let signature;
     try {
-      payload = JSON.parse(Buffer.from(parts[1], 'base64').toString());
-    } catch (e) {
-      response.status(400).end();
-      return;
-    }
-    if (payload.action === 'LOGOUT') {
-      let sessionIDs = payload.adapterSessionIds;
-      if (!sessionIDs) {
-        keycloak.grantManager.notBefore = payload.notBefore;
-        response.send('ok');
-        return;
-      }
-      if (sessionIDs && sessionIDs.length > 0) {
-        let seen = 0;
-        sessionIDs.forEach(id => {
-          keycloak.unstoreGrant(id);
-          ++seen;
-          if (seen === sessionIDs.length) {
+      signature = new Signature(keycloak.config);
+      signature.verify(token).then(token => {
```

Comment:
@abstractj It looks like we are not checking resource name. Is it correct?

Reply:
@mhajas I believe we handle this into the Grant Manager. Anyway, this PR is just a cherry-pick from what we already merged in the prod. So my suggestion is for us to get it merged and file a bug if there's anything missing.

@mhajas @hmlnarik that works for you?

Reply:
Hello, I tried to add this test:

````javascript
test('Test resource name', t => {
  t.plan(1);

  var app = new NodeApp();
  var client = admin.createClient(app.confidential('adminapp8'), realmName);

  return client.then((installation) => {
    app.build(installation);
    let opt = {
      method: 'post',
      url: `${app.address}/k_logout`,
      data: TestVector.logoutValidPayload
    };
    return axios(opt).then(response => {
      t.equal(response.status, 200);
    }).catch(err => {
      t.fail(err.response.data);
    });
  }).then(() => {
    app.destroy();
  });
});
````

I created the test like this: I took the valid scenario and changed the resource name of the deployment to something different (adminapp3 -> adminapp8). Should it be possible to logout from adminapp8 with token issued for resource adminapp3?